src/ResponseValidator.ts

```diff
@@ -95,20 +95,18 @@ export class ResponseValidator {
         // OpenID Connect Core 1.0 says that id_token is optional in refresh response:
         // https://openid.net/specs/openid-connect-core-1_0.html#RefreshTokenResponse
         if (response.isOpenId) {
-            if (response.id_token) {
-                this._validateIdTokenAttributes(response, state.id_token);
-                logger.debug("ID Token validated");
-            }
-            else {
-                // if there's no id_token on the response, copy over id_token from original request
-                response.id_token = state.id_token;
-                // and decoded part too
-                response.profile = state.profile;
-            }
+            this._validateIdTokenAttributes(response, state.id_token);
```

Comment:
I think this still needs to be wrapped with check if there actually is `response.id_token` to avoid throwing in https://github.com/authts/oidc-client-ts/blob/46fe10b1fe78f10d7764e900f19e98eed704d767/src/ResponseValidator.ts#L263